src/data/coinmasters.txt

```diff
@@ -1210,3 +1210,18 @@ Elf Guard Toy and Munitions Factory	buy	150	replica Elf Guard medal	ROW1428
 Elf Guard Toy and Munitions Factory	buy	400	Lil' Snowball Factory	ROW1429
 Elf Guard Toy and Munitions Factory	buy	1000	Elf Guard safety bear	ROW1440
 Elf Guard Toy and Munitions Factory	buy	2000	Elf Guard temporary (permanent) tattoo	ROW1430
+
+#Kiwi Kwiki Mart	unknown	mini kiwi bikini	mini kiwi (2)	ROW1477
```

Comment:
So we couldn't tell if it was a "buy" or a "sell", because we don't know that the currency of the Kiwi Kwiki Mart is a "mini kiwi", right? And we can't do that until the Coinmaster is implemented? I did that because the default USED to be "buy", and as a result, the Crimbo Armories - which had both "buy" and "sell" - autogenerated bogus entries for things which were "sell".

Huh.

The Coinmaster should be trivial, and re-visiting the mart after it is implemented should make these all "buy", I assume.

Reply:
I have not even hatched my mini kiwi egg yet, so have not even looked at the shop yet.
I am OK deferring coinmaster.txt until it is time to implement the Coinmaster - or mixing method.
I assume that the two ROWs for the mini kiwitini correspond to the bottle of gin and bottle of vodka variants.

Note that this is not a coinmaster with "multiple currencies" - although it presumably is a shop.php where some items take more than one item to make. Like Grandma. We do that via a mixing method.

Perhaps it is time to consider supporting "it's a coinmaster with a few recipes that take additional ingredients".

Reply:
Or perhaps we should just shrug and do it as a mixing method - existing technology which should handle this with ease.

My "unified shop.php class" which can be subclassed for CoinMasters, NPCStores, and MixingMethods can remain a pipe dream. :)

Reply:
To clarify my understanding, the current split is:
* if the shop uses meat, it's npcstores.txt
* if the shop uses exactly one currency per item, it's coinmasters.txt (e.g. Cosmic Ray's Bazaar)
* if the shop uses multiple currency items per item, it's concoctions (e.g. Mad Sliemce)
?

Reply:
I just saw this. Sorry!
That is exactly correct.

Reply:
The unified shop.php class deserves more than to be a pipe dream!